### ISSUE TITLE
Value drops to 0 after animation

### ISSUE THREAD

**Issue by isoengas** [ashish-chopra/ngx-gauge#20]
I have an strange effect where my gauge animation drops to 0 when the value is changed. I was able to reproduce the effect (I forked an existing stackblitz from another issue :)
https://stackblitz.com/edit/angular-n1vmyv?file=app%2Fapp.component.ts

The effect only occurs the first time when the value changes asynchronously from 0 to another value and it only affects the gauge bar, but not the text.

**Comment by ShuklaAmitKumar** [comment]
In constructor you are setting progress to "0".
   this.selectedBatch = {
      progress: 0
    }
So when your page load for the first time it's set value to "0". try to pass dynamic value

**Comment by isoengas** [comment]
After the first setTimeout call there is a discrepancy between the gauge label (85) and the bar (0), so I consider this a bug in the component. The stackblitz is just a simple case to demonstrate the bug, my case is quite different.

**Comment by ilivestrong** [comment]
Same issue I have.. Did anyone able to resolve it?

**Comment by nivu** [comment]
Me too facing this issue. Please update if you find any solution?

**Comment by nivu** [comment]
@ilivestrong I faced this issue in dev environment, but after production build, it is working fine. https://stackblitz.com/edit/angular-uhmnnt

**Comment by ilivestrong** [comment]
But what's the reason? @navneetnivu07

**Comment by dinohorvat** [comment]
Same problem... So if you have dynamic value (which everyone does), this is useless

**Comment by ashish-chopra** [comment]
Hello Everyone, 

Sorry for delaying it due to some personal reasons. But I have looked into the issue and realized that the issue is due to no synchronization between two cycles that are going on in the background:
1. Change Detection Cycle of Angular
2. RequestAnimationFrame Cycle(loop)

The problem is, whenever the value of gauge changes, it triggers change detection and invokes `_create()` method which initializes a new requestAnimationFrame loop from the current value of gauge to its new value. And it runs independently of CD cycle of angular. Imagine if there is another change to the gauge value while the first animation loop is in progress, so according to current code, the `_create()` will spawn another animation loop without killing the first one and then these two loops will run asynchronously and we cannot determine which one of those will finish later.

So the solution is to pre-emptively kill the animation loop (if it is already running) whenever the value changes before starting the new one. I have fixed this in my code, trying to test it and will soon push it. 

Also, another fix I want to add is to move animationLoop outside Angular using Zone.js which I will do later, once these fixes get stable.

Thanks, @isoengas for your pull request, it helped me to dive into it. And Thank you, everyone, for your discussion and stackblitz examples, it really fast-tracked my exploration.

Thanks!
Ashish
